### PR TITLE
Add Storybook to Seaqull | Closes #43

### LINKED ISSUE #43: Storybook
### What to do:
- [x] Installing storybook
- [x] Convert some components for testing matters
- [x] Add a wiki for storybook
_updated_
- [ ] Move storybook folders in 1 folder on root

### What is storybook
Shortly said, Stroybook is a sandbox environment for our individual components to check out if they work and how they look. You can edit them apart from the entire site saving trouble of reloading entire section when not needed to.

### PR DESCRIPTION
**Pull request by Samvdg**
Closes #43 

implemented storybook into the project

## How to run:
1. first do `npm ci`
2. run `npm run build-storybook` just to be safe
3. call `npm run storybook`
and now you should have the storybook page open

## What changed
- i added storybook to the project
- edited the storybook webpack.config.js so that it can read .sass files
- edited the storybook config.js for it to automaticly load in all .stories.js files

## Notice
- Test component stories will be removed at a later date
- Files in the folder `storybook-static` are automaticly generated

### CODE REVIEW

**Review by AjobK — CHANGES_REQUESTED** [review]
Works fine 👍 , though I'm not sure why but build-storybook doesn't work well?

Also, would be nice if you added a few dummy elements. I have merged develop so you should be able to.

File: .storybook/config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { configure } from '@storybook/react';
+
+// automatically import all files ending in *.stories.js
```
Comment (MiguelFieira) [inline .storybook/config.js]: Captial letter

File: .storybook/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// you can use this file to add your custom webpack plugins, loaders and anything you like.
```
Comment (MiguelFieira) [inline .storybook/webpack.config.js]: Unnecessary comments
Use capital letter at a start of a sentence

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// This is just the basic way to add additional webpack configurations.
+// For more information refer the docs: https://storybook.js.org/configurations/custom-webpack-config
+
+// IMPORTANT
+// When you add this file, we won't add the default configurations which is similar
```
Comment (MiguelFieira) [inline .storybook/webpack.config.js]: Unnecessary comments

File: .storybook/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// to "React Create App". This only has babel loader to load JavaScript.
+
+module.exports = {
+  plugins: [
+    // your custom plugins
```
Comment (MiguelFieira) [inline .storybook/webpack.config.js]: Plugins part can be removed...
You don't have any plugins

File: stories/index.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { linkTo } from '@storybook/addon-links';
+
+import { Button, Welcome } from '@storybook/react/demo';
+
+storiesOf('Welcome', module).add('to Storybook', () => <Welcome showApp={linkTo('Button')} />);
+
+storiesOf('Button', module)
+  .add('with text', () => <Button onClick={action('clicked')}>Hello Button</Button>)
```
Comment (MiguelFieira) [inline stories/index.js:12]: Remove the Example of storybook
Use a Seaqull component instead!

File: storybook-static/iframe.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
+<!doctype html><html><head><meta charset="utf-8"><title>Storybook</title><base target="_parent"><style>:not(.sb-show-main) > .sb-main,
```
Comment (MiguelFieira) [inline storybook-static/iframe.html:1]: Properly indent code!
It's a .html file

File: storybook-static/iframe.html [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  }</style><script>try {
+    if (window.parent !== window) {
+      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.parent.__REACT_DEVTOOLS_GLOBAL_HOOK__;
+      window.__VUE_DEVTOOLS_GLOBAL_HOOK__ = window.parent.__VUE_DEVTOOLS_GLOBAL_HOOK__;
+    }
+  } catch (e) {
+    console.warn('unable to connect to parent frame for connecting dev tools');
+  }</script></head><body><div class="sb-nopreview sb-wrapper"><div class="sb-nopreview_main"><h1 class="sb-nopreview_heading sb-heading">No Preview</h1><p>Sorry, but you either have no stories or none are selected somehow.</p><ul><li>Please check the storybook config.</li><li>Try reloading the page.</li></ul></div></div><div class="sb-errordisplay sb-wrapper"><div id="error-message" class="sb-heading"></div><pre class="sb-errordisplay_code">
+    <code id="error-stack"></code>
+  </pre></div><div id="root"></div><script src="runtime~main.2ce33a6eed47648b4286.bundle.js"></script><script src="vendors~main.2ce33a6eed47648b4286.bundle.js"></script><script src="main.2ce33a6eed47648b4286.bundle.js"></script></body></html>
```
Comment (MiguelFieira) [inline storybook-static/iframe.html]: Endline

File: storybook-static/index.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!doctype html><html><head><meta charset="utf-8"><title>Storybook</title><style>html, body {
```
Comment (MiguelFieira) [inline storybook-static/index.html:1]: Properly indent code!
It's a .html file

File: storybook-static/main.73f8654c8e52f28a85de.bundle.js [hunk 9]
```diff
@@ -0,0 +1 @@
+(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{288:function(n,o,p){n.exports=p(289)}},[[288,1,2]]]);
```
Comment (MiguelFieira) [inline storybook-static/main.73f8654c8e52f28a85de.bundle.js]: Endline

**Review by MiguelFieira — COMMENTED** [review]
See Comments

**Review by MiguelFieira — CHANGES_REQUESTED** [review]
See comments

File: storybook-static/iframe.html (hunk 6 above)
Comment (Samvdg) [reply storybook-static/iframe.html:1]: this file is from storybook itself. it's static

**Review by Samvdg — COMMENTED** [review]
@MiguelFieira The storybook-static/ files are automaticly generated. further more i fixed the rest of your comments that were of my own use

**Review by AjobK — CHANGES_REQUESTED** [review]
image

Hamburger looks messed up.

Also, did you check `npm run dev`?

**Review by AjobK — CHANGES_REQUESTED** [review]
Could you put everything storybook related in one folder?

For example .storybook and stories

File: stories/hamburger.stories.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import Hamburger from '../src/components/hamburger';
+
+import React from 'react';
+
+import { storiesOf } from '@storybook/react';
+// import { action } from '@storybook/addon-actions';
```
Comment (AjobK) [inline stories/hamburger.stories.js]: Remove these comments please

File: stories/index.js (hunk 5 above)
Comment (AjobK) [reply stories/index.js:12]: Agree

File: stories/hamburger.stories.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+// import { linkTo } from '@storybook/addon-links';
+
+storiesOf('Hamburger', module)
+	.add('Normal', () => <Hamburger/> )
+    .add('Clicked', () => <Hamburger active='true' /> )
```
Comment (AjobK) [inline stories/hamburger.stories.js]: Linting is terrible

File: stories/hamburger.stories.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (AjobK) [inline stories/hamburger.stories.js]: Because of linting it's requesting a new line

**Review by AjobK — CHANGES_REQUESTED** [review]
If you fix these changes I'll approve 👍

File: stories/index.js (hunk 5 above)
Comment (Samvdg) [reply stories/index.js:12]: will be removed later

**Review by MiguelFieira — APPROVED** [review]
If you remove the Example in the future then the rest is fine by me.

**Review by AjobK — APPROVED** [review]

**Pull request merged**